### ISSUE TITLE
Jenkins API not accessible(403 Forbidden error) with access token generated from azure ad

### ISSUE THREAD

**Issue by sanith01988** [jenkinsci/azure-ad-plugin#573]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
  Paste the output here
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

docker jenkins

### Reproduction steps

We're using azure ad app as security relam for our jenkins.Using the azure ad plugin through browser we're able to access the jenkins without any issue(We're not using graph api permission, instead our azure ad app assigned group configured via matrix authorization strategy). Now we need to call the jenkins api,for this we're having a technical account in entra and we're able to create access token for this user.But when we're using this access token for the jenkins api request we're getting 'Error accessing the API: 403 Client Error: Forbidden'.

### Expected Results

Jenkins api access with azure ad access token

### Actual Results

Error accessing the API: 403 Client Error: Forbidden

**Comment by timja** [comment]
I think you need Graph API permission for the API to work.

**Comment by sanith01988** [comment]
@timja Do  I need to set graph api permission to the app registration which is using in security relam of Jenkins.

**Comment by timja** [comment]
yes https://github.com/jenkinsci/azure-ad-plugin#setup-microsoft-entra-id-permissions-optional-but-recommended

> This is used for:
> * Jenkins looking up the user, e.g. when you use the Rest API

**Comment by sanith01988** [comment]
@timja 

I've enabled the entra id permission. Still I'm getting 403 error.
`403
<html><head><meta http-equiv='refresh' content='1;url=/securityRealm/commenceLogin?from=%2Fapi%2Fjson'/><script id='redirect' data-redirect-url='/securityRealm/commenceLogin?from=%2Fapi%2Fjson' src='/static/63c61884/scripts/redirect.js'></script></head><body style='background-color:white; color:white;'>
Authentication required


</body></html> ` In jenkins log I couldn't see any errors related to it.

**Comment by timja** [comment]
can you talk me through step by step what you've got setup?

**Comment by sanith01988** [comment]
@timja 

1. Set up jenkins with azure ad plugin as security relay using jenkins.yaml configured below details. 

jenkins:
  securityRealm:
    azureSecurityRealm:
      clientId: ''
      clientSecret: ''
      tenant: ''
      cacheduration: 0
      fromrequest: true

  authorizationStrategy:
    projectMatrix:
      entries:
        - group:
            name: <object-id>
            permissions:
              - Overall/Administer
        - user:
            name: <object-id>
            permissions:
              - Overall/Administer

2. Configured http:://<jenkins-name>/securityRealm/finishLogin as redirect uri in entra app registration
3. Added group & user to app registartion
4.Then throuh browser I'm able to access without any issue
5. then using a python script accessing the access_token with a technical user and using the access_ token try to access the "http:://<jenkins-name>/api/json"
token_endpoint = https://login.microsoftonline.com/<tenant_id>/oauth2/v2.0/token
response = requests.post(issuer["token_endpoint"], data={
            "client_id": '',
            'client_secret': '',
            'audience': ',
            'redirect_url': 'https://<jenkins-name>',
            "grant_type": "password",
            "username": 'technical_user',
            "password": 'password',
            "scope": "openid"
        })
access_token = response.json()['access_token']
url = "http://<jenkins-name>/api/json"
headers = {
    "Authorization": f"Bearer {access_token}", 
    "Accept": "application/json"
}

api_response = requests.get(url, headers=headers, verify=False)

**Comment by timja** [comment]
The Jenkins API uses a Jenkins API token and not a Microsoft access token.

Go to the Users profile in Jenkins and generate an API token

**Comment by laplace2108** [comment]
Having the same issue: I created a token as you suggested @timja, but got a 401. I can log in successfully in the web browser, but if I try to use the API with my SSO username and token, 401 is returned. Is this a limitation of the plugin?

**Comment by timja** [comment]
You need to use your object ID and not your SSO username
